All changes we discussed

Igor-Arena opened the pull request:
1. Changed files structure in the git:
- deleted duplicated files
- moved Api tests into test >  Api_tests folder
- created "Global_settings" folder in the "tests" folder for all global files
- moved my own files for studying into "automation_cource" folder
2. Changed Var to the Let for all tests
3. Names of variables are written from the small first letter according to the convention
4. Used global variables for api url 
5. Used "import..." in all files for import
6. Changed hardcoded data to the dynamic using "faker"
7. Removed "console.log" where its not needed

siarheima-papaya commented on tests/Api_tests/post.spec.ts:
```diff
@@ -0,0 +1,36 @@
+  ],
+  "status": "available"
+ };
+
```
add test.afterAll() and remove created pet. It necessary to remove test data after test launch. We always should clean up test data after tests

siarheima-papaya commented on tests/Api_tests/post.spec.ts:
```diff
@@ -0,0 +1,36 @@
+import {test, expect} from '@playwright/test'
```
All comments above valid for other tests. Please, fix them

siarheima-papaya commented on tests/Api_tests/delete.spec.ts:
```diff
@@ -0,0 +1,44 @@
+
+
+let petId: number;
+let randomFirstName = faker.person.firstName(undefined);
```
remove 'undefined'. You can use faker.person.firstName()

siarheima-papaya commented on package.json:
```diff
@@ -8,6 +8,7 @@
   "author": "",
```
Please, add script to run your tests

siarheima-papaya commented on tests/Api_tests/put.spec.ts:
```diff
@@ -0,0 +1,64 @@
+        "status": "available"}
+    });
+    console.log(response);
+    expect(response.status).toBe(200);
```
Add assertion to validate that data was changed

Igor-Arena commented:
1. Changed name of folders according to the convention
2. Added script
3. Added "Delete" api for tests
4. Deleted console.log
5. Deleted "undefined"

siarheima-papaya commented on tests/Global_settings/constant.ts:
```diff
@@ -0,0 +1 @@
+export let postAndGetPet = 'https://petstore.swagger.io/v2/pet/';
```
Rename to 'baseUrl'

siarheima-papaya commented on tests/Api_tests/get.spec.ts:
```diff
@@ -0,0 +1,47 @@
+});
+
+
+test.afterAll(async () => {
```
Move afterAll() after beforeAll

siarheima-papaya commented on tests/Api_tests/put.spec.ts:
```diff
@@ -0,0 +1,69 @@
+ };
+
+
+test('Post', async () => {
```
beforeAll

siarheima-papaya commented on tests/Api_tests/put.spec.ts:
```diff
@@ -0,0 +1,69 @@
+        ],
+        "status": "available"}
+    });
+    expect(response.status).toBe(200);
```
expect(res.data.category.name).toEqual(updatedFirstName)
expect(res.data.name).toEqual(updatedSecondName)

siarheima-papaya commented on tests/Api_tests/get.spec.ts:
```diff
@@ -41,7 +41,6 @@ test.afterAll(async () => {
     const response = await axios.get(`${baseUrl}${petId}`);
-    expect(response.data.id).toEqual(randomId);
     expect(response.data.category.name).toEqual(randomFirstName);
     expect(response.data.name).toEqual(randomSecondName);
```
Please, try to compare objects using `expect.objectContaining`. As I remember GET request return the same object as body for POST
```
{
  "id": 0,
  "category": {
    "id": 0,
    "name": "string"
  },
  "name": "doggie",
  "photoUrls": [
    "string"
  ],
  "tags": [
    {
      "id": 0,
      "name": "string"
    }
  ],
  "status": "available"
}
```

siarheima-papaya commented on playwright.config.ts:
```diff
@@ -10,7 +10,7 @@ import { defineConfig, devices } from '@playwright/test';
  */
 export default defineConfig({
-  testDir: './automation_cource',
+  testDir: './tests/Api_tests',
```
I see 2 directory: Api_tests and api_tests. 
Let's simplify renaming: create new directory `api_tests_pet_strore` -> move tests in it -> remove Api_tests and api_tests

siarheima-papaya commented on tests/Api_tests/post.spec.ts:
```diff
@@ -0,0 +1,46 @@
+    expect(response.data.id).toEqual(randomIdNumber);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
+    petId = response.data.id;
```
better to move this line directly after POST request
```
const response = await axios.post(baseUrl, postData);
petId = response.data.id;
...
```

siarheima-papaya commented on tests/Api_tests/put.spec.ts:
```diff
@@ -0,0 +1,77 @@
+test('PutRequest', async () => {
+    const response = await axios.put(baseUrl, 
+    { 
+      data: {
```
remove key 'data'. You should pass in just object, without `data:`
And then your test should works fine and response will contain names. So, it means that you can remove `test ('Get',....`

<img width="866" alt="Screenshot 2023-10-03 at 16 13 16" src="https://github.com/Igor-Arena/Demo-repo/assets/141643051/a34b6ab8-c51a-405f-a157-71eb54798468">

Igor-Arena commented:
1. Deleted folder Api_tests
2. Added folder api_tests_pet_store
3. Renamed folder Global_settings to global_settings
4. Deleted "data" keyword from the body in the "Put" test file
5. Deleted "Get" test from the "Put" test file
6. Moved "Petid" variable in the top of the test's body

siarheima-papaya commented on tests/api_tests_pet_store/get.spec.ts:
```diff
@@ -0,0 +1,49 @@
+test('Get', async () => {
+    const response = await axios.get(`${baseUrl}${petId}`);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
```
You can replace 44 and 45 lines using only one `expect.objectContaining()`. 
Please take a look at example https://playwright.dev/docs/api/class-genericassertions#generic-assertions-object-containing

siarheima-papaya commented on tests/api_tests_pet_store/post.spec.ts:
```diff
@@ -0,0 +1,46 @@
+    expect(response.status).toBe(200);
+    expect(response.data.id).toEqual(randomIdNumber);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
```
The same. You can replace by only one expect.objectContaining().

Igor-Arena commented:
1. Added expect.objectContaining
2. Added toMatchObject

siarheima-papaya commented on tests/api_tests_pet_store/delete.spec.ts:
```diff
@@ -0,0 +1,42 @@
+
+
+let postData = {
+  "id": randomId,
```
I just let you know, that it's possible to use key without quotes. 
E.q.
```
{
  id: randomId,
.....
 };
```

siarheima-papaya commented on tests/api_tests_pet_store/put.spec.ts:
```diff
@@ -0,0 +1,76 @@
+test.beforeAll(async () => {
+    const response = await axios.post(baseUrl, postData);
+    petId = response.data.id;
+    expect(response.status).toBe(200);
```
Please, remove assertion from beforeAll()

Igor-Arena commented:
Removed assertion from beforeAll in "Put" test
